legate/core/machine.py

```diff
@@ -168,6 +168,11 @@ def __eq__(self, other: object) -> bool:
                 return False
         return True
 
+    @property
+    def num_procs(self) -> int:
+        range = self.get_processor_range()
+        return range.high - range.low
```

Comment:
Let's not duplicate the size calculation, as we may later change the range back to a closed interval.
```suggestion
        return len(range)
```